Add matrix to build workflow

Aman123lug opened the pull request:
**What It Does**

Added a matrix to run the build across multiple OSes and Python versions #194

samadpls commented:
Hello @Aman123lug,

I've noticed that you're facing some challenges with test failures. I want to offer some suggestions:

- Instead of creating a new pull request on every change, you can clone the repository and push the changes directly.
- The reason you keep encountering build failures is that the current build only tests Python changes and not the build changes you made in the pull request.
- Regarding the changelog, I don't think you need to update it. @t1m0thyj can better guide you on whether the changelog needs to be updated or not.

Aman123lug commented:
@samadpls 
Thanks for your suggestions
I am new to this project and I need guidance can you please review my PR and tell me what I did wrong
If changelog does not require any update I will remove

Thank You!

t1m0thyj commented on CHANGELOG.md:
Thanks for updating the changelog 🙂  For this PR we don't need to update the changelog because we only do so when the changes affect end users (see [changelog update guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md#changelog-update-guidelines)).

Could you please remove the changelog update, since workflow changes aren't something that SDK users need to know about?

Aman123lug replied:
Okay i am removing the changelog Thank you

t1m0thyj commented on .github/workflows/sdk-build.yml:
```diff
@@ -8,14 +8,18 @@ on: [push, pull_request]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.7, 3.8, 3.9, 3.10, 3.11]
```
The Python versions need to be wrapped in quotes. Currently `3.10` is being treated as `3.1` since YAML thinks it is a floating point number that should be rounded.

t1m0thyj commented on .github/workflows/sdk-build.yml:
```diff
@@ -8,14 +8,18 @@ on: [push, pull_request]
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```
Please add `fail-fast: false` to the matrix definition so that if one build in the matrix fails, the others continue running (see [GitHub Actions docs](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures)).

Aman123lug replied:
Thanks for the feedback 
let me know, now it is correct or not 😅

t1m0thyj reviewed (changes requested):
@Aman123lug Thanks for contributing to the Zowe Python SDK 🙂 Left a few comments

t1m0thyj commented:
On line 28 of the workflow: `if [ -f requirements.txt ]; then pip install -r requirements.txt; fi`

The if condition written in Bash is unnecessary and will fail on Windows. I think this line could be replaced with just the pip command: `pip install -r requirements.txt`

Also, the PyYAML dependency in requirements.txt needs to be updated for newer versions of Python. I think we want to upgrade from version 5.4.1 to the latest one (6.0.1).

With these two changes, hopefully all the builds will pass 😋

samadpls commented:
Hey @Aman123lug , It appears that the `chanalog.md` file has been removed. Consequently, if this pull request is merged, it will result in the deletion of the `chanalog.md` file from the main branch as well.

Aman123lug commented:
Okay, then would you like me to create a new PR?
I changed the SDK-build.yml and added `pip install -r requirements.txt`
I also changed the PyYAML version latest one `6.0.1`

Aman123lug commented:
> Hey @Aman123lug , It appears that the `chanalog.md` file has been removed. Consequently, if this pull request is merged, it will result in the deletion of the `chanalog.md` file from the main branch as well.

Okay, Thanks for the review.
What should I do? Create new PR or change in Existing PR

samadpls commented:
> > Hey @Aman123lug , It appears that the `chanalog.md` file has been removed. Consequently, if this pull request is merged, it will result in the deletion of the `chanalog.md` file from the main branch as well.
> 
> Okay, Thanks for the review.
> What should I do? Create new PR or change in Existing PR
> 

I apologize for the confusion. Instead of creating a new pull request, please add the same 'changelog.md' file from this repository to your forked repository. Once you've done that, you'll be good to proceed.

Aman123lug commented:
> > > Hey @Aman123lug , It appears that the `chanalog.md` file has been removed. Consequently, if this pull request is merged, it will result in the deletion of the `chanalog.md` file from the main branch as well.
> > 
> > 
> > Okay, Thanks for the review.
> > What should I do? Create new PR or change in Existing PR
> 
> I apologize for the confusion😓. Instead of creating a new pull request, please add the same 'changelog.md' file from this repository to your forked repository. Once you've done that, you'll be good to proceed.

No worries. I created a new PR with all the same changes 😊
